### DISCUSSION TITLE
ChatGPT unable to read strings nested in array

### DISCUSSION THREAD

**Discussion by emilliepost** [AntlerVC/firetable#1186] [Support / Q&A] [answered]
Hi there folks,

I am trying to make chat gpt to summarize a text in json (array-object-string), but I cannot make it work. It is in the row Chapters.

Here is the Action script:

````
const action:Action = async ({row,ref,db,storage,auth,actionParams,user,logging}) => {
logging.log("action started")
const jsonResponse = row.Chapters;
  const res = await fetch("https://api.openai.com/v1/chat/completions", {
    method: "POST",
    headers: {
      "Content-Type": "application/json",
      "Authorization": "Bearer ***"
    },
    body: JSON.stringify({
    "model": "gpt-3.5-turbo",
    "max_tokens": 1500,
    "temperature": 0.9,
    "messages": [
        {
            "role": "user", "content": `summarize and paraphrase the following text:  ${row.Chapters}`
        }
    ]
})
  })
   const result = await res.json()
   await ref.update({ listOne: result })
  return {
    message: "Model Started",
    success: true
  }
   
}
````

I get a response from gpt but it keeps saying that there is no text. I tried "${row.Chapters}" and no luck.

Any ideas?

Thanks a lot in advance.

**Comment by devsgnr** [comment, marked as answer]
Hello @emilliepost - Rowy column names differ from the column key; Rowy automatically sets the key to a `lowercase/camelCase` of the column name. You want to be accessing the column key not the column name. Please can you confirm that your column key is correct? I'll leave an image reference where you can see your column key. See how column name **String** has a column key `string`.

<img width="404" alt="Screenshot 2023-03-29 at 19 13 31" src="https://user-images.githubusercontent.com/25407019/228631170-6b09eb3d-f84e-4ec3-9772-163e91dd56ac.png">

**Reply by emilliepost** [reply]
thanks for a swift reply 👌

I was not sure about this, since my key has Chapters not chapters
<img width="169" alt="Screenshot 2023-03-29 at 20 23 54" src="https://user-images.githubusercontent.com/72376544/228632708-7c3f1abb-7718-412c-be4a-81a82f05535f.png">
anyway, I tried lowercase and got the same response

**Reply by devsgnr** [reply]
Hey, @emilliepost. Is `row.Chapters` supposed to be a string or a json?

Just curious, because from this ChatGPT [documentation](https://platform.openai.com/docs/guides/chat#:~:text=OpenAI%20Playground.-,Introduction,-Chat%20models%20take), the message is sent as a string but `row.Chapters` seems to be a JSON field from your screenshot. 

Did you want to send an entire JSON object to OpenAI?

**Reply by emilliepost** [reply]
fixed it with a loop and extraction of properties, thanks for your help 🙏
